src/main/java/featurecat/lizzie/rules/SGFParser.java

```
@@ -367,4 +420,158 @@ private static String generateNode(Board board, BoardHistoryNode node) throws IO
 
     return builder.toString();
   }
+
+  public static boolean isListProperty(String key) {
+    for (String k : listProps) {
+      if (k.equals(key)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public static boolean isMarkupProperty(String key) {
+    for (String k : markupProps) {
+      if (k.equals(key)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Get a value with key, or the default if there is no such key
+   *
+   * @param key
+   * @param defaultValue
+   * @return
+   */
+  public static String optProperty(Map<String, String> props, String key, String defaultValue) {
+    return props.getOrDefault(key, defaultValue);
+  }
+
+  /**
+   * Add a key and value
+   *
+   * @param key
+   * @param value
+   */
+  public static void addProperty(Map<String, String> props, String key, String value) {
+    if (SGFParser.isListProperty(key)) {
+      // Label and add/remove stones
+      props.merge(key, value, (old, val) -> old + "," + val);
+    } else {
+      props.put(key, value);
+    }
+  }
+
+  /**
+   * Add the properties
+   *
+   * @return
+   */
+  public static void addProperties(Map<String, String> props, Map<String, String> addProps) {
+    if (addProps != null && addProps.size() > 0) {

```

When do you expect `addProps` to be `null`? It seems like it this the case then we have a bug, and I would be better to throw a NPE rather than simply ignoring the call.

---

Just remove the condition. Throw the NullPointException when has a except call.